app/services/digest_sender.rb

```diff
@@ -0,0 +1,48 @@
+class DigestSender
+  DIGEST_INTERVAL   = 1.day
+  INSTANT_INTERVAL  = 10.minutes
+
+  def self.send_digests
+    digest_users(type: :daily).each do |user|
+      DigestSender.new(user: user, type: :digest).send
+    end
+  end
+
+  def self.send_instants
+    digest_users(type: :instant).each do |user|
+      DigestSender.new(user: user, type: :instant).send
+    end
+  end
+
+  def self.digest_users(type:)
+    User.includes(:notifications).where("preferences LIKE '%digest_frequency: #{type}%'")
+  end
+
+  def initialize(type:, user:)
+    @type = type
+    @user = user
+  end
+
+  attr_accessor :type, :user
```

Comment:
Isn't the convention to have this right after the `class` line?

We've also used some other conventions to structure classes that I think it's worth keeping, including separating blocks of methods:

```ruby
  # -- Class Methods -----------------------------------------------------
```

and of course :alpha order.